cow_amm/cow_amm_lp_and_tvl_4047078.sql

```diff
@@ -0,0 +1,151 @@
+-- Computes the TVL, lp token total supply and lp token price of a CoW AMM pool over time
+-- Parameters
+--  {{token_a}} - either token of the desired uni pool
+--  {{token_b}} - other token of the desired uni pool
+--  {{start}} - date as of which the analysis should run
+
+-- Given that we might not have records every day in the source data (e.g. not every day the lp supply may change), 
+-- but still want to visualize development on a per day basis,  we create an auxiliary table with one record per 
+-- day between `start` and `now`
+with date_range as (
+    select t.day
+    from
+        unnest(sequence(
+            date(timestamp '{{start}}'),
+            date(now())
+        )) t (day) --noqa: AL01
+),
+
+-- Finds the CoW AMM pool address given tokens specified in query parameters (regardless of order) 
+cow_amm_pool as (
+    select
+        blockchain,
+        address,
+        token_1_address,
+        token_2_address
+    from query_3959044
+    where ((token_1_address = {{token_a}} and token_2_address = {{token_b}}) or (token_2_address = {{token_a}} and token_1_address = {{token_b}}))
```

Comment:
What happens here if there is more than one AMM per token pair?

Reply:
That should not be a problem, since we are not selecting a specific pool but rather a token pair, in the next steps we will sum all the tvl and lp supply for all these pools. This is actually the case with the token pair we are testing, so I find it find. Do you see any issues with such approach?

Reply:
Yes, I'm pretty sure the math doesn't check out when adding lp tokens across multiple pools, as lp tokens may have a different price per pool.

Take the following example
- pool a has $100 TVL and 100 LP tokens (1LP token = $1)
- pool b has $100 TVL and 10 LP tokens (1LP token = $10)

If we add this up, we would get $200 TVL for 110 LP tokens, making us believe on LP token is worth $1.8

Now let's say $10 are redeemed. Here it matters in which pool they are redeemed because:
- if redeemed in pool a, it will burn 10 LP tokens (leading to $190TVL and 100 LP tokens)
- if redeemed in pool b, it will burn 1LP token (leading to $190TVL and 109 LP tokens)

In each case we would end up with a different view of how much 1 LP token is worth.

So in summary, we need to fix ourselves to a specific pool (likely the biggest, for now I went with the most recently deployed one in the other query I wrote)